src/Symfony/Component/Translation/LocaleSwitcher.php

```diff
@@ -42,17 +50,26 @@ public function getLocale(): string
     /**
      * Switch to a new locale, execute a callback, then switch back to the original.
      *
-     * @param callable():void $callback
+     * @template T
+     *
+     * @param callable():T $callback
+     *
+     * @return T
      */
-    public function runWithLocale(string $locale, callable $callback): void
+    public function runWithLocale(string $locale, callable $callback): mixed
     {
         $original = $this->getLocale();
         $this->setLocale($locale);
 
         try {
-            $callback();
+            return $callback();
         } finally {
             $this->setLocale($original);
         }
     }
+
+    public function reset(): void
```

Comment:
Statefull services must at least reset their state.

Although the default locale is already set back in `LocaleAwareListener::onKernelFinishRequest()`, this doesn't cover eg message handlers.

Reply:
Do we need to implement `ResetInterface`?